api/pkg/import/zip_package_test.go

```diff
@@ -0,0 +1,228 @@
+
+	tempDir, err := ioutil.TempDir("", "test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	tempZipFile, err := ioutil.TempFile(tempDir, "test-archive*"+defaultImportArchiveExtension)
+	require.NoError(t, err)
+
+	err = writeZip(tempZipFile, sourceImportPackage)
+	require.NoError(t, err)
+
+	err = tempZipFile.Close()
+	require.NoError(t, err)
+
+	p, err := NewPackage(tempZipFile.Name(), testArchiveSize20MB)
+	require.NoError(t, err)
+	require.NotNil(t, p)
+
+	// assert that creating a package from a zipped file extracted the files in a subdir with the same name as the
+	// file minus the extension
+	expectedExtractedPath := strings.TrimSuffix(tempZipFile.Name(), defaultImportArchiveExtension)
+	assert.DirExists(t, expectedExtractedPath)
+	assertDirEqual(t, sourceImportPackage, expectedExtractedPath)
+
+	// assert that closing the package cleans up the extracted files
+	err = p.Close()
+	assert.NoError(t, err)
+	assert.NoDirExists(t, expectedExtractedPath)
+}
+
+func TestExtractErrorZipFilePackageTooBig(t *testing.T) {
+
+	sourceImportPackage := "../../test/data/import/sample-package"
+
+	tempDir, err := ioutil.TempDir("", "test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	tempZipFile, err := ioutil.TempFile(tempDir, "test-archive*"+defaultImportArchiveExtension)
+	require.NoError(t, err)
+
+	err = writeZip(tempZipFile, sourceImportPackage)
+	require.NoError(t, err)
+
+	err = tempZipFile.Close()
+	require.NoError(t, err)
+
+	p, err := NewPackage(tempZipFile.Name(), 10)
+	assert.ErrorIs(t, err, ErrorUncompressedSizeTooBig)
+	assert.Nil(t, p)
+
+	// the extraction folder is cleaned up
+	expectedExtractedPath := strings.TrimSuffix(tempZipFile.Name(), defaultImportArchiveExtension)
+	assert.NoDirExists(t, expectedExtractedPath)
+}
+
+func TestExtractErrorNonExistentZipFile(t *testing.T) {
+
+	tempDir, err := ioutil.TempDir("", "test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	nonExistingZipFileName := "thereisnospoon.zip"
+
+	p, err := NewPackage(path.Join(tempDir, nonExistingZipFileName), testArchiveSize20MB)
+	assert.Error(t, err)
+	assert.Nil(t, p)
+}
+
+func TestExtractErrorNoManifest(t *testing.T) {
+
+	sourceImportPackage := "../../test/data/import/invalid-package"
+
+	tempDir, err := ioutil.TempDir("", "test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	tempZipFile, err := ioutil.TempFile(tempDir, "test-archive*"+defaultImportArchiveExtension)
+	require.NoError(t, err)
+
+	err = writeZip(tempZipFile, sourceImportPackage)
+	require.NoError(t, err)
+
+	err = tempZipFile.Close()
+	require.NoError(t, err)
+
+	p, err := NewPackage(tempZipFile.Name(), testArchiveSize20MB)
+	assert.ErrorIs(t, err, os.ErrNotExist)
+	assert.Nil(t, p)
+
+	// the extraction folder is cleaned up
+	expectedExtractedPath := strings.TrimSuffix(tempZipFile.Name(), defaultImportArchiveExtension)
+	assert.NoDirExists(t, expectedExtractedPath)
+}
+
```

Comment:
I would add also a small test to check that we fail if an invalid zip is supplied

Reply:
There was a corresponding test for the handler before the refactoring, I will remove that and create a new one for the zip archive